app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -2321,4 +2323,29 @@ public function buildUrl($route, $routeParams = [], $absolute = true, $clickthro
 
         return $url;
     }
+
+    /**
+     * Check if an email is a variant by looking it up in emailSettings.
+     * Handles both parent emails and translations (translations inherit variant status from parent).
+     *
+     * @param array<int, array<string, mixed>> $emailSettings
+     */
+    private function isEmailVariant(int $emailId, array $emailSettings): bool
+    {
+        // Check if it's a parent email in emailSettings
+        if (isset($emailSettings[$emailId])) {
+            return (bool) $emailSettings[$emailId]['isVariant'];
+        }
+
+        // It's likely a translation - find it in translations and check if the parent is a variant
+        foreach ($emailSettings as $settings) {
+            if (!empty($settings['translations']) && is_array($settings['translations']) && isset($settings['translations'][$emailId])) {
```

Comment:
This will do the same, right? If the last condition is true then the first 2 must be true as well.
```suggestion
            if (isset($settings['translations'][$emailId])) {
```

Reply:
@escopecz I think if you dont check if it is empty (or isset) you will get a warning: 
PHP Warning:  Undefined array key "translations"

<?php
$settings = [];
echo "one:";
var_dump(is_array($settings['translations']));
echo "two:";
var_dump(!empty($settings['translations']) && is_array($settings['translations']));
?>

Reply:
```
<?php
$settings = [];
echo "one";
var_dump(is_array($settings['translations']));
echo "two";
var_dump(!empty($settings['translations']) && is_array($settings['translations']));
?>
```

Reply:
The `isset` doesn't throw warnings if any array key exists or not. Even if the `$settings` variable doesn't exist. See

https://legacy-sandbox.onlinephpfunctions.com?s=s7EvyCjg5eLlUknNTczM8UxRsFUwtgYJZKYpaGQWF6eWaKgAiZLMvPTiaPWSosS84pzEksz8vGL12GiYplhNTYVqXi4FIEhNzshXUEqtyCwuKVYCGlSrkJpTnIoqm5dfkAqWAwA%2C&v=8.2.20

It will return true only if all the array keys in the checked chain exist and the value is not NULL. So that's what we need.